ivy/func_wrapper.py

```diff
@@ -35,6 +35,175 @@
 # Helpers #
 # --------#
 
+# for casting modes, order is the hierarchy
+casting_modes_dict = {
```

Comment:
Can we get this dictionary dynamically instead? Would be easier to maintain if we just did 
```
{
    "uint": ivy.valid_uint_dtypes,
    "int": ivy.valid_int_dtypes,
    and so on ...
}
```

Reply:
sure, this would require us to sort the valid dtype dictionaries that we have maintained for each framework, because order is important here

Reply:
I think they are sorted already in ivy, which would save us from sorting explicitly 🙂

Reply:
I tried this, but due to cyclic dependency, we won't be able to use `ivy.valid_int/float/uint/dtypes` here in the func wrapper. Since this is a constant dictionary that won't require much update unless new types are found, we can continue with this

Reply:
Oh okay got it! I think we could also just not have this dictionary at all and simply make use of `ivy.valid_int_dtypes`, `ivy.valid_float_dtypes`, etc. in the casting functions directly. I think we'll always have those lists in sorted order anyway 🙂